src/analyses/apron/apronAnalysis.apron.ml

```diff
@@ -175,7 +206,14 @@ struct
 
   (* Function call transfer functions. *)
 
+  let pass_to_callee fundec reachable_from_args var =
+    let vname = Var.to_string var in
+    match List.find_opt (fun v -> v.vname = vname) (fundec.sformals @ fundec.slocals) with
```

Comment:
Now that Apron `Var`s for locals have their `vid` appended to their name, so `vname` has it, but `v.vname` from `fundec.slocals` does not. So does this ever find anything?

Reply:
Good catch! No, it doesn't, which was actually problematic, as too much was being passed to the caller. Changing it to take the `vid`s into account, we get the effect that if only some part of the locals is passed, relationality is lost between different locals of the caller. I now changed it such that if one local is reachable, all relevant locals are passed.